skimage/restoration/deconvolution.py

```diff
@@ -390,3 +391,137 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
         im_deconv[im_deconv < -1] = -1
 
     return im_deconv
+
+
+def blind_richardson_lucy(image, psf=None, iterations=10,
+                          return_iterations=False, clip=False):
+    """Blind Richardson-Lucy deconvolution.
+
+    Parameters
+    ----------
+    image : ndarray
+       Input degraded image (can be N dimensional).
+
+    psf : ndarray, optional
+       A first estimate of the point spread function, same size as image
+
+    iterations : int
+       Number of iterations. This parameter plays the role of
+       regularisation. After a given iterations, the estimates can produce
+       division by 0 problems, then the algorithm is automatically stopped.
+
+    return_iterations : boolean, optional
+        Returns instead of a tuple of the final restorated image and used PSF
+        a tuple of all iterations for further investigation
+
+    clip : boolean, optional
+       True by default. If true, pixel value of the result above 1 or
+       under -1 are thresholded for skimage pipeline compatibility.
```

Comment:
humm, this means that you make the assumption that the imput image dtype is float...

Reply:
I copied this from the original richardson lucy function, I convert it to float anyway in the beginning